db/keyval/etcd/config.go

```diff
@@ -96,6 +102,13 @@ func ConfigToClient(yc *Config) (*ClientConfig, error) {
 		}
 	}
 
+	if yc.Privatekeyfile != "" {
+		privateKey, err := keyval.ReadPrivateKey(yc.Privatekeyfile)
+		if err == nil {
+			cfg.PrivateKey = privateKey
```

Comment:
This should fail it error probably, since we failed to read private key we configured.